tests/ga/test_extension.py

```diff
@@ -2535,304 +2536,285 @@ def test_ext_handler_with_systemd_cgroup_enabled(self, *args):
+        second_ext = get_extension_actor(version="1.1.0")
 
-        with patch.object(CGroupConfigurator.get_instance(), "start_extension_command",
-                          side_effect=ExtensionError('disable Failed')) as patch_start_cmd:
-            with self.assertRaises(ExtensionError):
-                ExtHandlersHandler._update_extension_handler_and_return_if_failed(old_handler_i, new_handler_i)
+        TestExtensionUpdateOnFailure._do_upgrade_scenario(first_ext, second_ext)
 
-            args, kwargs = patch_start_cmd.call_args
+        first_ext.get_command("disableCommand").assert_called_once()
+        second_ext.get_command("installCommand").assert_called_once()
+        second_ext.get_command("enableCommand").assert_called_once()
 
-            self.assertTrue('-update' in kwargs['command'] and ExtCommandEnvVariable.DisableReturnCode in kwargs['env'],
-                            "The update command should have Disable Failed in env variable")
 
-    def test_uninstall_failed_env_variable_should_set_for_install_when_continue_on_update_failure_is_true(
-            self, *args):
-        old_handler_i = self._get_ext_handler_instance('foo', '1.0.0')
-        new_handler_i = self._get_ext_handler_instance('foo', '1.0.1', continue_on_update_failure=True)
+    def test_non_enabled_ext_should_not_be_disabled_at_ver_update(self):
 
-        with patch.object(CGroupConfigurator.get_instance(), "start_extension_command",
-                          side_effect=['ok', 'ok', ExtensionError('uninstall Failed'), 'ok']) as patch_start_cmd:
+        first_ext = get_extension_actor(enableAction=Actions.FailAction)
+        second_ext = get_extension_actor(version="1.1.0")
+
+        TestExtensionUpdateOnFailure._do_upgrade_scenario(first_ext, second_ext)
+
+        second_ext.get_command("installCommand").assert_called_once()
+        second_ext.get_command("enableCommand").assert_called_once()
+        first_ext.get_command("disableCommand").assert_not_called()
+        
 
-            ExtHandlersHandler._update_extension_handler_and_return_if_failed(old_handler_i, new_handler_i)
+    def test_disable_failed_env_variable_should_be_set_for_update_cmd_when_continue_on_update_failure_is_true(self):
+        exit_code = uuid.uuid4()    # Generate a unique UUID for highest strictness.
+        unique_fail = lambda *args, **kwargs: exit_code
 
-            args, kwargs = patch_start_cmd.call_args
+        first_ext = get_extension_actor(disableAction=unique_fail)
+        second_ext = get_extension_actor(version="1.1.0", continueOnUpdateFailure=True)
+
+        TestExtensionUpdateOnFailure._do_upgrade_scenario(first_ext, second_ext)
+
+        first_ext.get_command("disableCommand").assert_called_once()            
+        second_ext.get_command("updateCommand").assert_called_once()
+        first_ext.get_command("uninstallCommand").assert_called_once()
+        second_ext.get_command("installCommand").assert_called_once()
```

Comment:
I assume that at this point in time, when we want to assert if the command was called and how many times, we're past the point of checking if it suceeded? If the command failed (and it shouldn't have), we would get notified beforehand, correct?

Reply:
bump

Reply:
Sorry, didn't see this in the conversations tab.

Actually, since we're running the `run()` command, the `exthanders_handler` object would've swallowed any exceptions caused by the commands failing. Our only mechanism for checking a failed extension command is to check the `PUT /status` http endpoint (this is not unique to this PR's solution; rather, its intrinsic to the wholistic style of unit tests we have).

I hadn't thought to include a `ExtensionActor.status_blobs` check for every test, though there are some that have it. Do you think this would be a worthwhile addition to the tests?